sklearn/svm/src/liblinear/linear.cpp

```diff
@@ -2484,6 +2498,7 @@ model* train(const problem *prob, const parameter *param)
 		free(sub_prob.x);
 		free(sub_prob.y);
 		free(weighted_C);
+		free(sample_weight);
```

Comment:
Free'ing an array allocated with `new[]` is unsafe. You should either stick to `malloc`/`free`, or use `new[]`/`delete[]`, but never mix the two.